opendatahub-io/odh-dashboard
# Refactor ClusterSettings page and add storybook tests

Closes #951 

## Description


Refactor the codes in ClusterSettings.tsx and split them into different files based on the settings. No UI changes.

One tiny detailed change: Disable the submit button when the user sets the culler timeout to less than 10 mins (we only show the error message before but didn't disable the submit button)

## How Has This Been Tested?



1. Get an account with ODH admin access
2. Go to the cluster settings page of the dashboard
3. Change the settings, try to see how submit button performs and try to submit the changes to see if it works

## Test Impact


Add the storybook test for the whole page, since there is no e2e test for now, I can only test the user inputs and changes to check whether the error messages are correctly shown and check if the submit button is enabled/disabled based on the user inputs.
One thing that I addressed in this PR is I add this to all the tests.
```
configure({
  testIdAttribute: 'data-id',
});
```
We already set a lot of `data-id` in the frontend code to help with the jest test before. I think we should follow this way (compared to refactoring all the `data-id` to default `data-testid`).

## Request review criteria:
